python/meterpreter/tests/test_ext_server_stdapi.py

```diff
@@ -0,0 +1,168 @@
+default            10.79.0.1          UGScg          en0
+
+Internet6:
+Destination                             Gateway                         Flags         Netif Expire
+default                                 fe80::%utun0                    UGcIg         utun0
+""".lstrip()
+
+        process_mock = mock.Mock()
+        attrs = {
+            "stdout.read.return_value": popen_read_value,
+            "wait.return_value": ERROR_SUCCESS,
+        }
+        process_mock.configure_mock(**attrs)
+        mock_popen.return_value = process_mock
+
+        result = self.ext_server_stdapi[
+            "stdapi_net_config_get_routes_via_osx_netstat"
+        ]()
+
+        expected = [
+            {
+                "gateway": b"\nO\x00\x01",
+                "iface": "en0",
+                "metric": 0,
+                "netmask": b"\x00\x00\x00\x00",
+                "subnet": b"\x00\x00\x00\x00",
+            },
+            {
+                "gateway": b"\xfe\x80\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00",
+                "iface": "utun0",
+                "metric": 0,
+                "netmask": b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00",
+                "subnet": b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00",
+            },
+        ]
+
+        self.assertEqual(result, expected)
+
+    @mock.patch("subprocess.Popen")
+    def test_stdapi_net_config_get_interfaces_via_osx_ifconfig(self, mock_popen):
+        popen_read_value = b"""
+en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
+        options=400<CHANNEL_IO>
+        ether 11:22:33:44:55:66
+        inet 192.168.1.166 netmask 0xffffff00 broadcast 192.168.1.255
+        media: autoselect
+        status: active
+""".lstrip()
+
+        process_mock = mock.Mock()
+        attrs = {
+            "stdout.read.return_value": popen_read_value,
+            "wait.return_value": ERROR_SUCCESS,
+        }
+        process_mock.configure_mock(**attrs)
+        mock_popen.return_value = process_mock
+
+        result = self.ext_server_stdapi[
+            "stdapi_net_config_get_interfaces_via_osx_ifconfig"
+        ]()
+
+        expected = [
+            {
+                "addrs": [
+                    (
+                        socket.AF_INET,
+                        b"\xc0\xa8\x01\xa6",
+                        b"\xff\xff\xff\x00",
+                    )
+                ],
+                "flags": 8863,
+                "flags_str": "UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST",
+                "hw_addr": '\x11"3DUf',
+                "index": 0,
+                "mtu": 1500,
+                "name": "en0",
+            }
+        ]
+
+        self.assertEqual(result, expected)
+
+    def test_stdapi_net_config_get_interfaces(self):
+        request = bytes()
+        response = bytes()
+        self.assertErrorSuccess("stdapi_net_config_get_interfaces", request, response)
```

Comment:
I'll skip verifying the exact bytes for now, this was just an MVP "make sure it doesn't explode" test for now